IPython/utils/tests/test_tokenutil.py

```diff
@@ -29,65 +37,91 @@ def test_simple():
     for i in range(len(cell)):
         expect_token("foo", cell, i)
 
+
 def test_function():
     cell = "foo(a=5, b='10')"
-    # up to `foo(|a=`
-    for i in range(cell.find('a=') + 1):
-        expect_token("foo", cell, i)
-    # find foo after `=`
-    for i in [cell.find('=') + 1, cell.rfind('=') + 1]:
-        expect_token("foo", cell, i)
-    # in between `5,|` and `|b=`
-    for i in range(cell.find(','), cell.find('b=')):
+    for i in range(len(cell)):
         expect_token("foo", cell, i)
+    expect_token("a", "foo(|a, b)")
+    expect_token("foo", "foo(a|, b)")
+    expect_token("foo", "foo(a,| b)")
+    expect_token("b", "foo(a, |b)")
+    expect_token("foo", "foo(a, b|)")
 
-def test_multiline():
-    cell = '\n'.join([
-        'a = 5',
-        'b = hello("string", there)'
-    ])
-    expected = 'hello'
-    start = cell.index(expected) + 1
-    for i in range(start, start + len(expected)):
-        expect_token(expected, cell, i)
-    expected = 'hello'
-    start = cell.index(expected) + 1
-    for i in range(start, start + len(expected)):
-        expect_token(expected, cell, i)
 
-def test_multiline_token():
-    cell = '\n'.join([
-        '"""\n\nxxxxxxxxxx\n\n"""',
-        '5, """',
-        'docstring',
-        'multiline token',
-        '""", [',
-        '2, 3, "complicated"]',
-        'b = hello("string", there)'
-    ])
-    expected = 'hello'
+@pytest.mark.parametrize(
+    "cell",
+    [
+        "\n".join(["a = 5", "b = hello('string', there)"]),
+        "\n".join(
+            [
+                '"""\n\nxxxxxxxxxx\n\n"""',
+                '5, """',
+                "docstring",
+                "multiline token",
+                '""", [',
+                '2, 3, "complicated"]',
+                'b = hello("string", there)',
+            ]
+        ),
```

Comment:
I'm fine with `\n.join`; have you thought on textwrap.dedent for readability ?

Reply:
This is copied over from another test that had the same assertions but that's a good idea